src/CouchDB.Driver/Settings/CouchSettings.cs

```diff
@@ -29,6 +30,14 @@ public interface ICouchConfiguration
         /// <returns>The current settings</returns>
         ICouchConfiguration UseCookieAuthentication(string username, string password, int cookieDuration = 10);
 
+        /// <summary>
+        /// Enables proxy authentication. 
+        /// </summary>
+        /// <param name="username">Server username.</param>
+        /// <param name="roles">Server roles.</param>
+        /// <returns>The current settings</returns>
+        ICouchConfiguration UseProxyAuthentication(string username, IReadOnlyCollection<string> roles, string? token = null);
```

Comment:
I prefer a simple IEnumerable here or IList, because otherwise people need to create a readonly collection instead of just `new[] { "role1", "role2" }`

Reply:
AFAIK `T[]` implements `IReadOnlyCollection<T>`.

Reply:
True.. Bad example... I mean why limit to IReadOnlyCollection?

Reply:
`IList` is broader. `IEnumerable` does not have `Count`. The implementation might change that passing empty `roles` will omit the whole header.

Reply:
IList is fine if you agree

Reply:
Why `IList`? The implementation will never need to modify the collection. Moreover, users should be sure that this lib will not modify it. Try to limit the contract (interface) on minimum what make sense. (`IEnumerable` might be too little. `IList` is too much.)

Reply:
Just because most be people use `List<T>` for ease to use.
But I get your point